source/tutorials/nix-language.md

```diff
@@ -0,0 +1,1679 @@
+
+The purpose of the Nix language is to create *build tasks*: precise descriptions of how contents of existing files are used to derive new files.
+
+:::{important}
+A build task in Nix is called *derivation*.
+:::
+
+The Nix language has only few basic constructs which can be combined arbitrarily:
+
+- Primitive data types
+
+  as basic building blocks
+
+- Compound data types and, in the broader sense, functions
+
+  to produce and transform complex data
+
+- Name assignment
+
+  to manipulate complex data as units
+
+In addition it allows one significant side effect:
+reading files, to capture what build tasks will operate on.
+
+There is nothing else to it.
+What may look complicated comes not from the language, but from how it is used.
+
+(names-values)=
+## Names and values
+
+Values in the Nix language can be primitive data types, lists, attribute sets, and functions.
+
+We show primitve data types and lists as examples in the context of [attribute sets](attrset).
+Later in this section we cover special features of character strings: [file system paths](file-system-paths), [antiquotation](antiquotation), and [indented strings](indented-strings).
+We deal with [functions](functions) separately.
+
+There are two ways to assign names to values in Nix: attribute sets and `let` expressions.
+
+Assignments are denoted by a single equal sign (`=`).
+
+(attrset)=
+### Attribute set `{ ... }`
+
+An attribute set is an unordered collection of name-value-pairs, where names must be unique.
+
+The following example shows all primitive data types, lists, and attribute sets.
+
+:::{note}
+If you are familiar with JSON, imagine the Nix language as *JSON with functions*.
+
+Nix language data types *without functions* work just like their counterparts in JSON and look very similar.
+:::
+
+
+<table>
+<tr>
+  <th>Nix language</th>
+  <th>JSON</th>
+</tr>
+<tr>
+<td>
+
+```nix
+{
+  string = "hello";
+  integer = 1;
+  float = 3.141;
+  bool = true;
+  null = null;
+  list = [ 1 "two" false ];
+  attribute-set = {
+    a = "hello";
+    b = 2;
+    c = 2.718;
+    d = false;
+  }; # comments are supported
+}
+```
+
+</td>
+<td>
+
+```json
+{
+  "string": "hello",
+  "integer": 1,
+  "float": 3.141,
+  "bool": true,
+  "null": null,
+  "list": [1, "two", false],
+  "object": {
+    "a": "hello",
+    "b": 1,
+    "c": 2.718,
+    "d": false
+  }
+}
+```
+
+</td>
+</tr>
+</table>
+
+:::{note}
+- Attribute names usually do not need quotes.[^1]
+- List elements are separated by white space.[^2]
+:::
+
+[^1]: Details: Nix manual - attribute naming rules <!-- TODO: create and link manual section -->
+[^2]: Details: [Nix manual - lists][manual-lists]
+
+[manual-lists]: https://nixos.org/manual/nix/stable/language/values.html#list
+
+(rec-attrset)=
+#### Recursive attribute set `rec { ... }`
+
+You will sometimes see attribute sets declared with `rec` prepended.
+This allows access to attributes from within the set.
```

Comment:
I don't go into dynamic attributes because they are rare enough in practically relevant code not to matter here. If you just learn about Nix and want to go through some tutorials here you won't dig into Nixpkgs internals or NixOS module implementations where those become more prominent.